sw/device/tests/rstmgr_alert_info_test.c

```diff
@@ -254,29 +257,33 @@ static alert_info_t alert_info_dump2struct(
   myinfo.class_accum_cnt[3] = (uint16_t)(upper | lower);
   // loc_alert_cause
   myinfo.loc_alert_cause = (dump[6] >> 12) & 0x7f;
-  // alert_cause
-  lower = dump[7] & 0xffff;
-  upper = dump[7] >> 16;
 
+  // alert_cause
   // dump[6] layout:
   //  31                     19,
   // |--- alert_cause[12:0]---+
   // 18                    12,11 0
   // +--loc_alert_cause[6:0]--+--class_accum_cnt[3][15:4]--|
   //
-  // total 16 + 13 lower bits are valid
-  lower = (lower << 13) + (dump[6] >> 19);
 
-  // dump[8] lower16 + dump[7] upper 16
-  upper = upper + ((dump[8] & 0xffff) << 16);
+  // the lower 32-bit of the alert cause is made up of:
```

Comment:
there was a minor bug with this one before, so the upper 29 bits or so wouldn't collect correctly.
I think we should move this into a testutil for rstmgr probably.